test/e2e/README.md

```diff
@@ -16,10 +16,28 @@ or
 make local-e2e
 ```
 or
-```sh
+```bash
 go test -tags "e2e all_providers" -run [test name regex]
 ```
 
+### Configuring OS version for the tests
+
+In each provider-specific file in the [E2E framework folder](../framework), there are helper methods of the form `With<OS family><KubeVersion>()`, for example, `WithUbuntu127()`, which instructs the tests to use an Ubuntu template/image corresponding to Kubernetes 1.27, depending on the provider. These methods take an OS version as an optional function parameter to construct the template/image name or environment variable. If the OS family you are writing tests for has multiple versions, for example, Ubuntu 20.04 and Ubuntu 22.04, and you do not provide an OS version is provided when writing the test, the [default version](../framework/os_versions.go) defined for the OS will be used to configure the template/image. If an OS version is provided, that will be used instead.
+
+For example, here is how you would write vSphere Kubernetes 1.27 simple flow test that uses an Ubuntu 22.04 template.
+```Go
+func TestVSphereKubernetes127Ubuntu2204SimpleFlow(t *testing.T) {
+	test := framework.NewClusterE2ETest(
+		t,
+		framework.NewVSphere(t, framework.WithUbuntu127(framework.Ubuntu2204Version)),
+		framework.WithClusterFiller(api.WithKubernetesVersion(v1alpha1.Kube127)),
```

Comment:
I love we documenting this. However, could we favor using `provider.WithKubeVersionAndOS` instead of provider options?

I would prefer all new tests to start using this as we slowly move towards refactoring the old ones. This pattern is more flexible and it favors parametrized tests.

Reply:
Do you mean changing the function signature of `New<Provider>` methods to not have provider options in it?

Reply:
Ah nono
More like this
```go
func TestVSphereKubernetes127Ubuntu2204SimpleFlow(t *testing.T) {
	vsphere := framework.NewVSphere(t)
	test := framework.NewClusterE2ETest(t, vsphere)
	test.WithClusterConfig(
		vsphere.WithKubeVersionAndOS(v1alpha1.Kube127, framework.Ubuntu2204),
	)

	runSimpleFlow(test)
}
```

Reply:
changed! Thank you for the example @g-gaston!